pkg/cloud/google/machineactuator.go

```diff
@@ -796,19 +800,51 @@ func getOrNewComputeServiceForMachine(params MachineActuatorParams) (GCEClientCo
 	if params.ComputeService != nil {
 		return params.ComputeService, nil
 	}
-	// The default GCP client expects the environment variable
-	// GOOGLE_APPLICATION_CREDENTIALS to point to a file with service credentials.
-	client, err := google.DefaultClient(context.TODO(), compute.ComputeScope)
-	if err != nil {
-		return nil, err
+
+	var client *http.Client
+	var err error
+	// If specified in the GCE config, use the alternative authentication.
+	if params.CloudConfigPath != "" {
+		glog.Info("Trying to get open the GCE config")
+		client, err = clientWithAltTokenSource(params.CloudConfigPath)
 	}
+	if params.CloudConfigPath == "" || err != nil {
+		glog.Info("Using the default GCP client")
+		if err != nil {
+			glog.Warningf("Could not create an alternative auth client: %q", err)
+		}
+		// The default GCP client expects the environment variable
+		// GOOGLE_APPLICATION_CREDENTIALS to point to a file with service credentials.
+		client, err = google.DefaultClient(context.TODO(), compute.ComputeScope)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	computeService, err := clients.NewComputeService(client)
 	if err != nil {
 		return nil, err
 	}
 	return computeService, nil
 }
 
+func clientWithAltTokenSource(gceConfigPath string) (*http.Client, error) {
+	glog.Info("Trying to get the alt token")
+	gceConfig := struct {
+		Global struct {
+			ProjectID string `gcfg:"project-id"`
+			TokenURL  string `gcfg:"token-url"`
+			TokenBody string `gcfg:"token-body"`
+		}
+	}{}
+	if err := warnings.FatalOnly(gcfg.ReadFileInto(&gceConfig, gceConfigPath)); err != nil {
```

Comment:
Why not `gcfg.FatalOnly` to avoid the extra import? (That call just proxies to `warnings.FatalOnly` anyway)

I ask because that's how it's done [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce.go#L277).

Reply:
I copied over from here: https://github.com/kubernetes/cloud-provider-gcp/blob/1aa2dddbbc427899e0dd5f313829552c0892e247/cmd/gcp-controller-manager/app/options.go#L112

Switching to gcfg.FatalOnly

Reply:
Should I remove the vendored stuff now?

Reply:
If you re-run `dep` does it remove any vendored code taking out this import?

Reply:
You can also run `dep status` to ensure that your dependencies are all properly accounted for.